Description: Add extra argument to enable building of JAR files
 The JAR file is shipped in an Architecture: all package so only needs to be
 built once. This patch adds an extra argument to enable/disabling building of
 the JAR file seperatly from the BD-Java support.
Author: Sebastian Ramacher <sramacher@debian.org>
Forwarded: not-needed
Last-Update: 2015-05-03

--- a/configure.ac
+++ b/configure.ac
@@ -89,6 +89,12 @@
 AC_ARG_ENABLE([udf],
   [AS_HELP_STRING([--enable-udf], [enable UDF support @<:@default=disabled@:>@])])
 
+AC_ARG_ENABLE([bdjava-jar],
+  [AS_HELP_STRING([--enable-bdjava-jar],
+  [enable JAR file for BD-Java support (default is no)])],
+  [use_bdjava_jar=$enableval],
+  [use_bdjava_jar=no])
+
 AC_ARG_WITH([libxml2],
   [AS_HELP_STRING([--without-libxml2], [build without libxml2 support @<:@default=with@:>@])])
 
@@ -224,7 +230,7 @@
   ])
 
   AC_CHECK_PROG(HAVE_ANT, [ant], yes, no)
-  if test "x$HAVE_ANT" = "xno"; then
+  if test "x$use_bdjava_jar" = "xyes" && test "x$HAVE_ANT" = "xno"; then
       AC_MSG_ERROR([BD-J requires ANT, but ant was not found. Please install it.])
   fi
 
@@ -233,6 +239,7 @@
   AC_DEFINE_UNQUOTED([JDK_HOME], ["$JDK_HOME"], [""])
 fi
 AM_CONDITIONAL([USING_BDJAVA], [ test $use_bdjava = "yes" ])
+AM_CONDITIONAL([USING_BDJAVA_BUILD_JAR], [ test $use_bdjava_jar = "yes" ])
 
 dnl BD-J type
 if test "$BDJ_TYPE" = "j2me"; then
@@ -292,6 +299,7 @@
 echo "  BD-J support:                  $use_bdjava"
 if [[ $use_bdjava = "yes" ]]; then
 echo "  BD-J type:                     $BDJ_TYPE"
+echo "  build JAR:                     $use_bdjava_jar"
 if test x"$BDJ_BOOTCLASSPATH" != x""; then
 echo "  BD-J bootclasspath:            $BDJ_BOOTCLASSPATH"
 fi
--- a/Makefile.am
+++ b/Makefile.am
@@ -186,10 +186,12 @@
 
 
 if USING_BDJAVA
+if USING_BDJAVA_BUILD_JAR
 jardir=$(datadir)/java/
 jar_DATA=$(top_builddir)/.libs/libbluray-$(BDJ_TYPE)-$(VERSION).jar
 
 $(top_builddir)/.libs/libbluray-$(BDJ_TYPE)-$(VERSION).jar: all-local
+endif
 
 libbluray_la_SOURCES += \
 	src/libbluray/bdj/bdj.h \
@@ -214,6 +216,7 @@
 
 AM_CFLAGS += $(BDJAVA_CFLAGS)
 
+if USING_BDJAVA_BUILD_JAR
 all-local:
 	ant -f $(top_srcdir)/src/libbluray/bdj/build.xml \
 	    -Dbuild='$(abs_builddir)/src/libbluray/bdj/build' \
@@ -229,6 +232,7 @@
 	    -Dversion='$(BDJ_TYPE)-$(VERSION)' \
 	    clean
 endif
+endif
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = src/libbluray.pc
